### PR TITLE
feat: Remove support for MongoDB 4.0

### LINKED ISSUE #8272: Remove support for MongoDB 4.0
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation

MongoDB 4.0 reached its EOL in April 2022. 

### Feature / Enhancement Description

Remove support for MongoDB 4.0:
- Remove related tests (restricted to MongoDB version)
- Remove from CI workflow

### Example Use Case
n/a

### Alternatives / Workarounds
n/a

### LINKED ISSUE #8273: Replace standalone MMAPv1 CI test and replace with standalone WT
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation

Deprecated in MongoDB 4.0, the MMAPv1 storage engine and all its related flags and options are removed in MongoDB 4.2.

### Feature / Enhancement Description

- Replace standalone MMvP CI test and replace with standalone WT
- Remove "WiredTiger" references from CI workflows and test code as it becomes the default

### Example Use Case
n/a

### Alternatives / Workarounds
n/a

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description

Removes support for mongodb 4.0

Related issue: #8272, https://github.com/parse-community/parse-server/issues/8273
Closes #8272 
Closes https://github.com/parse-community/parse-server/issues/8273

### Approach

Removes support for mongodb 4.0

### TODOs before merging

- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

**Comment by dblythy** [comment]
I cannot find any tests that use `it_only_mongodb_version` for v4.0

**Comment by dblythy** [comment]
Can this and #8273 be resolved in the same PR?

**Comment by mtrezza** [comment]
Yes, because https://github.com/parse-community/parse-server/issues/8273 is only a CI issue and it has a dependency to this issue here.

**Review by mtrezza — CHANGES_REQUESTED** [review]
~~Could you remove the conflicts?~~

Done

File: package.json [hunk 1]
```diff
@@ -112,20 +112,20 @@
     "lint-fix": "eslint --fix --cache ./",
     "build": "babel src/ -d lib/ --copy-files",
     "watch": "babel --watch src/ -d lib/ --copy-files",
-    "test:mongodb:runnerstart": "cross-env MONGODB_VERSION=${MONGODB_VERSION:=$npm_config_dbversion} MONGODB_TOPOLOGY=${MONGODB_TOPOLOGY:=standalone} MONGODB_STORAGE_ENGINE=${MONGODB_STORAGE_ENGINE:=wiredTiger} mongodb-runner start",
-    "test:mongodb:testonly": "cross-env MONGODB_VERSION=${MONGODB_VERSION:=$npm_config_dbversion} MONGODB_TOPOLOGY=${MONGODB_TOPOLOGY:=standalone} MONGODB_STORAGE_ENGINE=${MONGODB_STORAGE_ENGINE:=wiredTiger} TESTING=1 jasmine",
+    "test:mongodb:runnerstart": "cross-env MONGODB_VERSION=${MONGODB_VERSION:=$npm_config_dbversion} MONGODB_TOPOLOGY=${MONGODB_TOPOLOGY:=standalone} mongodb-runner start",
+    "test:mongodb:testonly": "cross-env MONGODB_VERSION=${MONGODB_VERSION:=$npm_config_dbversion} MONGODB_TOPOLOGY=${MONGODB_TOPOLOGY:=standalone} TESTING=1 jasmine",
     "test:mongodb": "npm run test:mongodb:runnerstart --dbversion=$npm_config_dbversion && npm run test:mongodb:testonly --dbversion=$npm_config_dbversion",
     "test:mongodb:4.0.28": "npm run test:mongodb --dbversion=4.0.28",
```
Comment (mtrezza) [inline package.json]:
```suggestion
```

**Review by mtrezza — APPROVED** [review]
Looks good!

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-alpha.6](https://github.com/parse-community/parse-server/releases/tag/6.0.0-alpha.6)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/6.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [6.0.0](https://github.com/parse-community/parse-server/releases/tag/6.0.0)

**Pull request merged**